### PR TITLE
Add gettext libraries

### PR DESCRIPTION
**Pull request by oleastre**
Prepare work to solve the internationalization problem

### CODE REVIEW

**Comment by MoNTE48** [comment]
https://github.com/MultiCraft/MultiCraft/tree/master/lib/intl

**Comment by oleastre** [comment]
I only compiled the _runtime_ part of gettext, which is quite lightweight and self contained. And as minetest already includes and rely on gettext it seemed to be the best fit.
Now, if another library if preferred, feel free to tell me. 
But I also did a PR in minetest with the required changes to get the translation working on android.

**Comment by oleastre** [comment]
Rebased on latest master.

File: Android/Build/Makefile [hunk 1]
```diff
@@ -604,6 +610,60 @@ $(LUAJIT_LIB) : $(LUAJIT_TIMESTAMP)
...
+	fi
+
+gettext_download :
+	@if [ ! -d ${GETTEXT_DIR} ] ; then									\
+		echo "gettext sources missing, downloading...";					\
+		mkdir -p ${ANDR_ROOT}/deps;										\
+		cd ${ANDR_ROOT}/deps ;											\
+		wget ${GETTEXT_URL} || exit 1;									\
+		tar zxf gettext-${GETTEXT_VERSION}.tar.gz || exit 1;					\
+		$(RM) -r gettext-${GETTEXT_VERSION}.tar.gz || exit 1;					\
```
Comment (MoNTE48) [inline Android/Build/Makefile:564]: 2 extra tabs here and on above line (625, 626)
Same on line 656.

File: Android/Build/Makefile [hunk 2]
```diff
@@ -115,6 +115,12 @@ LUAJIT_TIMESTAMP = $(LUAJIT_DIR)/timestamp
 LUAJIT_TIMESTAMP_INT = $(ANDR_ROOT)/deps/luajit_timestamp
 LUAJIT_URL_HTTP = https://github.com/LuaJIT/LuaJIT/archive/v$(LUAJIT_VERSION).zip
 
+GETTEXT_VERSION = 0.20.2
```
Comment (MoNTE48) [inline Android/Build/Makefile]: 0.21.0 is latest version (confirmed: works!)

**Comment by Pevernow** [comment]
@sfan5 What do you think of this issue?
Would you like to merge it?

**Comment by Pevernow** [comment]
@oleastre Please rebase this pull request, thanks.

**Comment by oleastre** [comment]
Done, but I did not tested the compilation nor updated the binaries to 0.21.0, although I updated the Makefile.

**Review by sfan5 — COMMENTED** [review]
The Makefile mess should be going away instead of us having more of it but I don't care at this point.

**Pull request merged**